src/test/scala/com/microsoft/hyperspace/index/HybridScanForIcebergTest.scala

```diff
@@ -0,0 +1,171 @@
+import org.apache.iceberg.hadoop.HadoopTables
+import org.apache.iceberg.spark.SparkSchemaUtil
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRelation}
+import scala.collection.JavaConverters._
+
+import com.microsoft.hyperspace.{Hyperspace, TestConfig}
+
+class HybridScanForIcebergTest extends HybridScanSuite {
+  override protected val fileFormat = "iceberg"
+  override protected val fileFormat2 = "iceberg"
+
+  override def beforeAll() {
+    super.beforeAll()
+    spark.conf.set(
+      "spark.hyperspace.index.sources.fileBasedBuilders",
+      "com.microsoft.hyperspace.index.sources.iceberg.IcebergFileBasedSourceBuilder," +
+        "com.microsoft.hyperspace.index.sources.default.DefaultFileBasedSourceBuilder")
+    hyperspace = new Hyperspace(spark)
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    spark.conf.unset("spark.hyperspace.index.sources.fileBasedBuilders")
+  }
+
+  override def setupIndexAndChangeData(
+      sourceFileFormat: String,
+      sourcePath: String,
+      indexConfig: IndexConfig,
+      appendCnt: Int,
+      deleteCnt: Int): (Seq[String], Seq[String]) = {
+    // Use partition to control the number of appended and deleted files.
+    val dfFromSampleRenamed = dfFromSample.withColumnRenamed("Date", "date")
+    val icebergTable = createIcebergTable(sourcePath, dfFromSampleRenamed, Seq("RGUID"))
+    dfFromSampleRenamed.write.format(sourceFileFormat).mode("overwrite").save(sourcePath)
+    val df = spark.read.format(sourceFileFormat).load(sourcePath)
+    hyperspace.createIndex(df, indexConfig)
+    icebergTable.refresh()
+    val inputFiles = icebergTable
+      .currentSnapshot()
+      .addedFiles()
+      .asScala
+      .toSeq
+      .map(f => s"${f.path().toString}")
+
+    // Create a new version by deleting entries.
+    if (deleteCnt > 0) {
+      val filesToDelete = inputFiles.dropRight(inputFiles.size - deleteCnt)
+      val deleteAction = icebergTable.newDelete()
+      filesToDelete.foreach(deleteAction.deleteFile)
+      deleteAction.commit()
+    }
+    if (appendCnt > 0) {
+      dfFromSampleRenamed
+        .limit(appendCnt)
+        .write
+        .format(sourceFileFormat)
+        .mode("append")
+        .save(sourcePath)
+    }
+
+    icebergTable.refresh()
+
+    val inputFilesA = inputFiles.map(f => s"file:/${f.replaceAll("^/", "")}")
+    val inputFilesB = icebergTable
+      .newScan()
+      .planFiles()
+      .iterator()
+      .asScala
+      .toSeq
+      .map(f => s"file:/${f.file().path().toString.replaceAll("^/", "")}")
+    (inputFilesB diff inputFilesA, inputFilesA diff inputFilesB)
+  }
+
+  test(
+    "Append-only: filter rule & parquet format, " +
+      "index relation should include appended file paths.") {
+    // This flag is for testing plan transformation if appended files could be load with index
```

Comment:
Sorry I didn't revised this comment after refactoring 😅.

```suggestion
    // This test is for verifying plan transformation if appended files could be load with index
```